project/src/Enemies/Pirate/PirateShip.gd

```diff
@@ -94,3 +134,53 @@ func _update_agent() -> void:
 	agent.linear_velocity.x = _velocity.x
 	agent.linear_velocity.y = _velocity.y
 	agent.angular_velocity = _angular_velocity
+
+
+func _set_behaviors_on_distances() -> void:
+	var distance_from_spawn := agent.position.distance_to(spawn_location.position)
+
+	if distance_from_spawn > distance_from_spawn_max:
+		_arrive_home_blend.is_enabled = true
+		_pursue_face_blend.is_enabled = false
+	else:
+		if player_agent:
+			var distance_from_player := agent.position.distance_to(player_agent.position)
+			if distance_from_player < aggro_radius:
+				_pursue_face_blend.is_enabled = true
+				_arrive_home_blend.is_enabled = false
+
+
+func _set_firing_on_player() -> void:
+	if not player_agent:
+		return
+	
+	if _pursue_face_blend.is_enabled:
+		var to_player := (
+				Vector2(
+						agent.position.x,
+						agent.position.y
+				) -
+				Vector2(
```

Comment:
How about adding helper methods or objects in the steering toolkit to work on 2D games? I see a lot of manual conversion from V3 to V2 in your code. Razvan added a utility function for AStar in our 2D project, as Godot's AStar works in 3D by default:

```gdscript
Utils.to_vector2(point)
```

Reply:
Yeah, I thought about it. That and a "Kinematic2DSteeringAgent" that would automatically update itself instead of having to maintain an _update_agent function.

I'll add some issues on the toolkit for consideration.